[CI] EsIndexSerializationTests testEqualsAndHashcode failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3740 / centos-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/wpqj2nmvnoncc)
- [elasticsearch-periodic-platform-support #3692 / almalinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/ozmcadqphkn3c)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.index.EsIndexSerializationTests.testEqualsAndHashcode" -Dtests.seed=3BD254C1CC6CC972 -Dtests.locale=en-IE -Dtests.timezone=Brazil/Acre -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testEqualsAndHashcode'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.index.EsIndexSerializationTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: EsIndex mutation should not be equal to original
Expected: not <uWHMu>
     but: was <uWHMu>
```

**Issue Reasons:**
- [main] 2 failures in test testEqualsAndHashcode (0.3% fail rate in 769 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (11.8% fail rate in 17 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/index/EsIndexSerializationTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/index/EsIndexSerializationTests.java
@@ -58,8 +58,8 @@ public class EsIndexSerializationTests extends AbstractWireSerializingTestCase<E
     @Override
     protected EsIndex mutateInstance(EsIndex instance) throws IOException {
         String name = instance.name();
-        Map<String, EsField> mapping = randomMapping();
-        Set<String> concreteIndices = randomConcreteIndices();
+        Map<String, EsField> mapping = instance.mapping();
+        Set<String> concreteIndices = instance.concreteIndices();
         switch (between(0, 2)) {
             case 0 -> name = randomValueOtherThan(name, () -> randomAlphaOfLength(5));
             case 1 -> mapping = randomValueOtherThan(mapping, EsIndexSerializationTests::randomMapping);

